internal/kubernetes/autoscaler.go

```diff
@@ -74,4 +74,37 @@ func scaleUpDeployment(item v1.Deployment, namespace string, clientDeployment v1
 	wg.Done()
 }
 
+func ScalingEngine() {
+	log.Info().Msgf("Starting Scaling Engine")
+	clientDeployment := clientSet.AppsV1().Deployments(config.Namespace)
+	labelSelector := metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("%s=%s", config.LabelProxlessEnabled, "true"),
+	}
+	for true {
+		deploys, err := clientDeployment.List(labelSelector)
+		if err != nil {
+			log.Error().Err(err).Msgf("Could not list deployments with label %s=true in namespace %s", config.LabelProxlessSvc, config.Namespace)
+			// don't do anything else, we don't wanna kill the proxy
+		} else {
+			for _, item := range deploys.Items {
+				label := item.Labels[config.LabelProxlessSvc]
+				if route, err := store.GetRouteByLabel(label); err != nil {
+					log.Error().Err(err).Msgf("Could not get route from map for deployment with label %s=true in namespace %s", config.LabelProxlessSvc, config.Namespace)
+				} else if *item.Spec.Replicas > int32(0) {
+					timeIdle := time.Now().Sub(route.LastUsed).Seconds()
+					if timeIdle >= float64(config.ServerlessTTL) {
+						item.Spec.Replicas = int32Ptr(0)
+						if _, err := clientDeployment.Update(&item); err != nil {
+							log.Error().Err(err).Msgf("Could not scale down the deployment %s in namespace %s", item.Name, item.Namespace)
+						} else {
+							log.Debug().Msgf("Deployment %s in namespace %s scaled down after %s", item.Name, item.Namespace, config.ServerlessTTL)
+						}
+					}
+				}
+			}
+		}
+		time.Sleep(time.Duration(config.ServerlessPollInterval) * time.Second)
```

Comment:
This is gonna be great with something like Celery / tasks base system once u work on the scaling.